website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,325 @@
+var $ = require('jquery');
+var m = require('mithril');
+var moment = require('moment');
+var $osf = require('js/osfHelpers');
+var LogText = require('js/logTextParser');
+var pips = require('sliderPips');
+
+var xhrconfig = function (xhr) {
+    xhr.withCredentials = true;
+    xhr.setRequestHeader('Content-Type', 'application/vnd.api+json');
+    xhr.setRequestHeader('Accept', 'application/vnd.api+json; ext=bulk');
+
+};
+
+var LogWrap = {
+    controller: function(args){
+        var self = this;
+        self.userId = args.userId;
+        self.activityLogs = m.prop();
+        self.eventFilter = false;
+        self.dateEnd = moment.utc();
+        self.dateBegin = moment.utc();
+        self.today = moment.utc();
+        self.sixMonthsAgo = moment.utc().subtract(6, 'months');
+        self.page = 1;
+        self.cache = [];
+        self.loading = false;
+        self.div = 8.64e+7;
+        self.canvasHeight = 40;
+        self.totalEvents = 0;
+        self.eventNumbers = {};
+
+        self.getLogs = function(init, reset, update) {
+            if (!(init || reset || update)  && self.cache[self.page - 1]){
```

Comment:
It would here, but the rest of the function would have a bunch of string comparisons, like:
```
if ((state === 'init') || (state === 'update')) {
     ...
}
if (state !== 'init') {
    ...
}
```